pkg/cmd/cli/cmd/deploy.go

```diff
@@ -323,6 +323,11 @@ func (o DeployOptions) retry(config *deployapi.DeploymentConfig) error {
 	return nil
 }
 
+// nowFn is extracted from the cancel method for unit testing.
+var nowFn = func() time.Time {
```

Comment:
Suggestion: instead of a global function, we can make this an unexported field of `DeployOptions` that by default is implemented with `unversioned.Now().Time`, but we can change in tests without mutating global state.